Core/Session/IGrindingSession.cs

```diff
@@ -0,0 +1,23 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace Core.Session
+{
+    public interface IGrindingSession
```

Comment:
Can we rename everywhere `IGrindingSession` -> `IGrindSession`. `Grinding` sounds a little bit weird for me